### PR TITLE
Version 18 release duplicate key issue

### PR DESCRIPTION
**Pull request by angelozerr**
Version 18 release duplicate key issue

Fixes https://github.com/redhat-developer/vscode-quarkus/issues/822

### CODE REVIEW

**Comment by fbricon** [comment]
tests are failing, better fix those before merging

**Comment by datho7561** [comment]
I confirmed that I was able to reproduce the bug using the sample project, and this PR fixes that bug. I'm going to try running the test suite locally to see if I the results are any different locally.

**Comment by angelozerr** [comment]
> tests are failing, better fix those before merging

I change the strategy by checking if parameter exists before to add it.

**Comment by angelozerr** [comment]
> I confirmed that I was able to reproduce the bug using the sample project, and this PR fixes that bug. I'm going to try running the test suite locally to see if I the results are any different locally.

Thanks to review it.

I have changed my code, it should work better now (I hope).

**Comment by angelozerr** [comment]
CI build is working again and it fixes a test that I had to updated.

I will try to write an another test.

@datho7561 please retry it.

File: qute.jdt/com.redhat.qute.jdt.test/src/main/java/com/redhat/qute/jdt/template/TemplateGetDataModelProjectTest.java [hunk 1]
```diff
@@ -436,7 +436,7 @@ public void checkedTemplateWithDefaultName() throws Exception {
 		List<DataModelParameter> helloParameters = helloTemplate.getParameters();
 		Assert.assertNotNull(helloParameters);
 
-		Assert.assertEquals(2, helloParameters.size());
+		Assert.assertEquals(1, helloParameters.size());
```
Comment (angelozerr) [inline qute.jdt/com.redhat.qute.jdt.test/src/main/java/com/redhat/qute/jdt/template/TemplateGetDataModelProjectTest.java:439]: The PR fixes a bug with the test which uses:

```java
@CheckedTemplate(defaultName=CheckedTemplate.HYPHENATED_ELEMENT_NAME)
	static class Templates2 {
		static native TemplateInstance HelloWorld(String name);

	}

```

JDT Java search engine reports 2 search matches:

 * one for CheckedTemplate
 * one for CheckedTemplate.HYPHENATED_ELEMENT_NAME

which explains the duplicate.

/cc @rgrunber perhaps it is a bug of JDT?

My fix  is just checking that parameter exists before add it.

**Review by datho7561 — APPROVED** [review]
Looks good. Thanks Angelo!

**Pull request merged**